src/rust/entrystore/src/seg/mod.rs

```diff
@@ -43,16 +43,30 @@ impl Seg {
 
         // build the datastructure from the config
         let data = ::seg::Seg::builder()
+            .restore(config.restore())
             .hash_power(config.hash_power())
             .overflow_factor(config.overflow_factor())
             .heap_size(config.heap_size())
             .segment_size(config.segment_size())
             .eviction(eviction)
             .datapool_path(config.datapool_path())
+            .segments_fields_path(config.segments_fields_path())
+            .ttl_buckets_path(config.ttl_buckets_path())
+            .hashtable_path(config.hashtable_path())
             .build();
 
         Self { data }
     }
+
+    /// Flush (gracefully shutdown) the `Seg` cache if configured to do so
+    pub fn flush<T: SegConfig>(self, config: &T) {
+        let config = config.seg();
+
+        if config.graceful_shutdown() {
+            // TODO: check if successfully shutdown and record result
+            self.data.flush();
```

Comment:
This line currently produces a warning as I am ignoring the Result. Ideally the result would be intepreted here or passed along to `server/segcache/src/lib.rs` where I intend this flush() function to be called